docs/source/api/usim.rst

```diff
@@ -1,19 +1,163 @@
-.. this API reference is autogenerated by apidoc
-   TODO: replace it with a manually curated one
+μSim API Reference
+==================
 
-usim package
-============
+.. py:module:: usim
+    :synopsis: The μSim namespace
 
-.. automodule:: usim
-    :members:
-    :undoc-members:
-    :show-inheritance:
+The :py:mod:`usim` package provides access to the entire μSim API in one
+convenient, flat namespace.
+Logically, the API can be divided into different topics, however.
+In addition, :py:mod:`usim.typing` allows for type annotations to statically
+verify simulations.
 
-Submodules
-----------
+.. hint::
+
+    μSim provides a :ref:`compatibility layer <simpy_compatibility>`
+    to the `SimPy`_ simulation framework.
+    The :py:mod:`usim.py` package is a drop-in replacement
+    for the :py:mod:`simpy` package.
+
+Starting a Simulation
+---------------------
+
+Simulations in μSim are always defined based on a number of initial root
+:term:`activities <activity>`,
+which may branch out to more :term:`activities <activity>`.
+A simulation is started by calling :py:func:`usim.run`, passing in its
+:term:`activities <activity>` and optionally the time window to simulate.
+
+.. autofunction:: usim.run
+
+Simulating Time
+---------------
+
+μSim provides a number of primitives to check or wait for specific points in time
+of a simulation.
+
+:py:data:`usim.time`
+    Expressions define points in time or delays.
+    For example, ``await (time + 20)`` delays for 20 time units.
+
+:py:data:`usim.eternity`
+    A point in time indefinitely far into the future
+
+:py:data:`usim.instant`
+    A point in time from the current moment.
+
+In addition, recurring actions can be repeated at specific delays or intervals
+in ``async for`` block.
+
+:py:data:`usim.interval`
+    Repeat in fixed intervals, regardless of any time spent in a block.
+
+:py:data:`usim.delay`
+    Repeat after fixed delays, in addition to any time spent in a block.
+
+Branching and Multitasking
+--------------------------
+
+An :term:`activity` may not only ``await`` another,
+it may also run one or more :term:`activities <activity>` concurrently.
+This requires opening a scope in an ``async with`` context
+which defines the lifetime of child activities.
+
+:py:class:`usim.Scope`
+    An :term:`asynchronous context manager` which can concurrently
+    run :term:`activities <activity>` until it is closed.
+    Forcefully closed if an exception occurs
+    in the hosting :term:`activity` or a child.
+
+:py:func:`usim.until`
+    A :py:class:`~usim.Scope` that is also forcefully closed
+    when a specific notification triggers.
+
+Each child :term:`activity` is represented by a :py:class:`~usim.typing.Task`.
+These can be inspected for their current status,
+and various exceptions are unique to them.
```

Comment:
Are the exceptions unique for the current status?
What do you think about making it into two sentences. It might help to state:

```suggestion
Various exceptions are unique to tasks.
```